include/seastar/core/semaphore.hh

```diff
@@ -154,6 +154,7 @@ public:
 private:
     ssize_t _count;
     std::exception_ptr _ex;
+    bool _used;
```

Comment:
Shouldn't we do in debug mode only like the reverted patch did?

Reply:
makes sense